src/passes/DeadCodeElimination.cpp

```diff
@@ -321,84 +322,62 @@ struct DeadCodeElimination : public WalkerPass<PostWalker<DeadCodeElimination>>
     }
   }
 
-  void visitSetLocal(SetLocal* curr) {
-    if (isUnreachable(curr->value)) {
-      replaceCurrent(curr->value);
+  // Append the reachable operands of the current node to a block, and replace
+  // it with the block
+  void BlockifyReachableOperands(std::vector<Expression*> list, WasmType type) {
+    for (size_t i = 0; i < list.size(); ++i) {
```

Comment:
fair point, it could in theory be used to operate on a list with more than Index elements.